### PR TITLE
Master merge for #915

### PR DESCRIPTION
**Pull request by swizzlr**
#915

### CODE REVIEW

**Comment by alblue** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
I am wondering if we should have a test case for this bug fix. I'll check if the `HTTPServer.swift` can be extended to simply reflect back the request headers for an incoming request.

**Comment by pushkarnk** [comment]
@swizzlr It will be great if we can have a test on these lines (feel free to use this code, if needed):
https://github.com/pushkarnk/swift-corelibs-foundation/commit/2b1e93ba10b7f512ffaf6577f9612aa8a08d44e5

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
@swift-ci please test

**Comment by pushkarnk** [comment]
Unrelated failure. Let me try one more time.

@swift-ci please test

**Comment by pushkarnk** [comment]
Same compiler crash again. May be we should wait?
```
Failing Tests (1):
    Swift(linux-x86_64) :: compiler_crashers/28719-currentconstraintsolverarena-no-constraint-solver-active.swift
```

**Comment by swizzlr** [comment]
@pushkarnk Well, this crash doesn't have anything to do with this PR, right?

**Comment by pushkarnk** [comment]
Yes, but it can't be merged without the CI tests passing :( Let me try again.

@swift-ci please test and merge

**Comment by pushkarnk** [comment]
@swift-ci please test and merge

**Comment by swizzlr** [comment]
Compiler engineers, right?
